src/libraries/System.Net.Http/src/System/Net/Http/MetricsHandler.cs

```diff
@@ -0,0 +1,231 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Diagnostics.Metrics;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.Net.Http;
+
+internal sealed class MetricsHandler : HttpMessageHandlerStage, IRequestFailureMetricsLogger
+{
+    private readonly HttpMessageHandler _innerHandler;
+    private readonly Meter _meter;
+    private readonly UpDownCounter<long> _currentRequests;
+    private readonly Counter<long> _failedRequests;
+    private readonly Histogram<double> _requestsDuration;
+
+    public MetricsHandler(HttpMessageHandler innerHandler, Meter meter)
+    {
+        _meter = meter;
+        _innerHandler = innerHandler;
+
+        _currentRequests = _meter.CreateUpDownCounter<long>(
+            "http-client-current-requests",
+            description: "Number of outbound HTTP requests that are currently active on the client.");
```

Comment:
Shall we localize this?

Reply:
I'm going to guess no. Event source display names weren't localized - https://github.com/dotnet/runtime/blob/89f3a9ef41383bb409b69d1a0f0db910f3ed9a34/src/libraries/System.Net.Http/src/System/Net/Http/HttpTelemetry.AnyOS.cs#L54